File: 0367-valid-perfect-square/0367-valid-perfect-square.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        int l = 0, r = num;
        long ans = 0 , mid = 0;
        while(l <= r){
            mid = l + (r-l)/2;
            if(mid*mid == num){
                return true;
            }
            if(mid*mid < num){
                ans = mid;
                l = mid+1;
            }
            else{
                r = mid-1;
            }

        }
        return false;
    }
};

File: 0367-valid-perfect-square/0367-valid-perfect-square.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        l ,r = 0, num
        mid = 0
        while (l <= r):
            mid = l + (r-l)//2
            if(mid*mid == num):
                return True
            if(mid*mid < num):
                l = mid+1
            elif(mid*mid > num):
                r = mid -1
        return False